### ISSUE TITLE
[Feature] Introduce Binding DSL

### ISSUE THREAD

**Issue by nbrugger-tgm** [nbrugger-tgm/reactj#42]
### Why introducing now?

It is a breaking change as the API will be completely different, and as the switch to V4 (modularization) already contains many major changes and reworks it is better to release now than later.

### What would the DSL (Domain Specific Language) look like?

Not specified yet, something like 
```java
public createBindings(ReactiveBinder<Person> binder){
    binder.call(textfield::setText).with(Person::getName);
    binder.call(Person::setName).with(textfield::getText).when(textChanges(textfield));

    //or
    binder.bind(textfield::setText).to(convert(Person::getAge).with(Integer::toString));

    //binder
    binder.set(textfield:setText).to(Person::getName);
}
```

### Operations

- [x] Converting
- [x] Method reference use
- [x] Reacting to Model property changes
- [x] Conditional bindings
- [x] ~dual-way bindings~ (not required anymore due to DSL structure) 

### Concept : Lists
(Most likely needs an internal `diff` operation)
```java
binder.list(Person::getAddresses)
    .onAdd(addressPanel::add)
    .onRemove(addressPanel::remove)
    .useComponent(AddressView::new)
```
- ~List as Model~
- ~List as Property of Model~
- [x] List Diff
- [x] List Diff compliant proxy

### Further requirements

- [x] Unit-test coverage > 95%
- [x] Every public method documented

**Comment by nbrugger-tgm** [comment]
## Grammer/syntax design:
![binding_dsl](https://user-images.githubusercontent.com/34156963/137595869-baffbd20-c631-4525-9f0a-5be40bf6d252.png)

The white triangles are the entry points

**Comment by nbrugger-tgm** [comment]
## Status update

https://github.com/nbrugger-tgm/reactj/pull/49#issuecomment-945280768

**Comment by nbrugger-tgm** [comment]
Blocked by #53

**Comment by nbrugger-tgm** [comment]
With completion of #53 and #44 #54 a big leap was done.

The DSL is completed and implemented for all modules, the missing thigs that will be implemented  in #49 are 

- [ ] > 70% coverage
- [ ] > 85% coverage
- [ ] JavaDoc for all new interfaces in API module
- [ ] Fixing the 100rd minors/issues/codesmells
